subt/control_center_qt.py

```diff
@@ -0,0 +1,283 @@
+
+    def _createStatusBar(self):
+        self.status = self.statusBar()
+
+    def _createDock(self):
+        d = QDockWidget("All Robots", self)
+        w = QWidget()
+        layout = QVBoxLayout()
+        layout.setAlignment(Qt.AlignTop)
+        layout.addWidget(QPushButton("Pause Mission", clicked=self.on_pause_mission))
+        layout.addWidget(QPushButton("Continue Mission", clicked=self.on_continue_mission))
+        layout.addWidget(QPushButton("Exit Mission", clicked=self.on_stop_mission))
+        w.setLayout(layout)
+        d.setWidget(w)
+        d.setDisabled(True)
+        self.addDockWidget(Qt.RightDockWidgetArea, d)
+
+    def on_start_stop(self, state):
+        if self.recorder is None:
+            assert state == Qt.Checked
+            self.recorder = record(self.centralWidget(), self.cfg)
+            self.cc = next(self.recorder)
+            print("recording started")
+        else:
+            assert state == Qt.Unchecked
+            with suppress(StopIteration):
+                next(self.recorder)
+            self.recorder = None
+            print("recording stopped")
+        for d in self.findChildren(QDockWidget):
+            d.setDisabled(self.recorder is None)
+
+    def closeEvent(self, e):
+        self.centralWidget().close()
+
+    def on_pause_mission(self):
+        print("pause mission")
+        self.cc.pause_mission()
+
+    def on_continue_mission(self):
+        print('continue mission')
+        self.cc.continue_mission()
+
+    def on_stop_mission(self):
+        print("stop mission")
+        self.cc.stop_mission()
+
+
+def record(view, cfg):
+    with osgar.logger.LogWriter(prefix='control-center-') as log:
+        log.write(0, bytes(str(cfg), 'ascii'))
+        recorder = osgar.record.Recorder(config=cfg['robot'], logger=log)
+        cc = recorder.modules['cc']
+        cc.view = view
+        view.robot_statuses = {}
+        with recorder:
+            yield cc
+
+
+class View(QWidget):
+
+    robot_status = pyqtSignal(int, list, str)
+    show_message = pyqtSignal(str, int)
+    colors = [QColor(Qt.white), QColor(Qt.green), QColor(Qt.blue), QColor(Qt.red),
+              QColor(Qt.cyan), QColor(Qt.magenta), QColor(Qt.yellow)]
+
+    def __init__(self):
+        super().__init__()
+        self.robot_status.connect(self.on_robot_status)
+        self.setAutoFillBackground(True)
+        p = self.palette()
+        p.setColor(self.backgroundRole(), Qt.black)
+        self.setPalette(p)
+        self.robot_statuses = {}
+
+    def on_robot_status(self, robot, pose2d, status):
+        self.show_message.emit(f"received position {pose2d} and status {status} from robot {robot}", 3000)
+        self.robot_statuses.setdefault(robot, []).append((pose2d, status))
+        self.update()
+
+    def paintEvent(self, e):
+        t = QTransform() # world transform
+        t.translate(self.width() / 2, self.height() / 2) # in pixels
+        t.scale(0.1, -0.1) # 1px == 1cm (position is in mm)
```

Comment:
was not here 0.01? I see I did not realized that the position was not converted to meters after was received.

Reply:
The scale 0.01 results in 1px == 10cm. If the position was converted to meters first, than scale 0.01 would result in 1px == 100m. Now the code and comment is in sync and this also happens to correspond with the default scale in lidarview.